ucsb-cs156-s24/proj-happycows-s24-4pm-6
# Admin can remove user from commons - FRONTEND

## Overview
In this PR, I finished adding the frontend of a new feature. It adds on a column in the Users table, only viewable by an admin, which displays "Edit Commons." When clicking it, an admin can see a check box list of all of the commons that specific user is a part of. The admin can then click on all of the commons they would like to remove the user from and click "Remove from selected commons." This edited code also incorporates a bug fix from Dakota which allows for the removing a user form a commons API call to work correctly. The files to checkout are UsersTable.js and UsersTable.test.js. ProfitsRepository.js includes the bug fix. ProfitRepository.java includes Dakota's bug fix. The remaining changed files are from my previous pull request for the backend.

## Screenshots (Optional)
<img width="792" alt="Screenshot 2024-05-21 at 3 04 43 PM" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-6/assets/89555340/4886573f-d8e5-4396-899e-637aea1c4b27">
<img width="1173" alt="Screenshot 2024-05-21 at 3 05 03 PM" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-6/assets/89555340/a6b84871-a951-4dbe-a5a6-fb7562cde1de">
<img width="1142" alt="Screenshot 2024-05-21 at 3 05 12 PM" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-6/assets/89555340/71ebebeb-77f7-4663-ac9f-18cf24cc2d92">


## Validation (Optional)
1. Download this branch.
2. Run spring-boot.
3. Login as an admin and join the available commons.
4. Navigate to "Users" and click "Edit Commons" next to your name.
5. Remove yourself from the commons you just joined and see that you are no longer in them on the home page.

## Tests


## Linked Issues
Closes #12
